feat(janitor): add Nebius Cloud (MK8s) CSP support for node reboot operations

nv-oviya opened the pull request:
## Summary

Adds Nebius AI Cloud as a supported Cloud Service Provider (CSP) for the NVSentinel Janitor module, enabling automated node reboot operations on Nebius Managed Kubernetes (MK8s) clusters.

## Motivation

NVSentinel users running GPU workloads on Nebius AI Cloud currently cannot use the Janitor module for automated fault remediation. This PR extends CSP support to Nebius, allowing the Janitor to reboot nodes with failing GPUs on MK8s clusters.

## Changes

### New Files

- **`janitor/pkg/csp/nebius/nebius.go`**
  - Implements `model.CSPClient` interface for Nebius
  - Uses official Nebius Go SDK (`github.com/nebius/gosdk`) for API interactions
  - Supports authentication via:
    - Service account key file (`NEBIUS_SA_KEY_FILE`)
    - Direct IAM token (`NEBIUS_IAM_TOKEN`)
    - Workload identity (SDK auto-discovery)
  - Implements stop/start pattern for reboot (Nebius has no direct reboot API)
  - Parses Nebius MK8s provider ID format: `nebius://computeinstance-{id}`

### Modified Files

- **`janitor/pkg/csp/client.go`**
  - Added `ProviderNebius` constant
  - Registered Nebius client in `NewWithProvider()` switch
  - Added Nebius to `GetProviderFromString()` for case-insensitive matching

- **`janitor/pkg/csp/client_test.go`**
  - Added test cases for Nebius provider string parsing
  - Added Nebius to provider constants validation
  - Added Nebius to `NewWithProvider` test matrix

- **`distros/kubernetes/nvsentinel/charts/janitor/values.yaml`**
  - Added `nebius` to supported provider types
  - Added `nebius` configuration block with auth options

## Implementation Details

### Reboot Mechanism

Nebius Compute API does not expose a direct "reboot" operation. This implementation uses:

1. **Stop** - `instanceService.Stop()` halts the instance
2. **Wait** - Waits for stop operation to complete
3. **Cooldown** - 5-minute cooldown period
4. **Start** - `instanceService.Start()` restarts the instance
5. **Ready check** - Polls instance state until `RUNNING`

### Provider ID Parsing

Nebius MK8s uses the provider ID format:
```
nebius://computeinstance-e00a2rcz1xvbqgbvp5
```

The parser handles:
- `nebius://instance-id` (primary format)
- `nebius:///instance-id` (triple slash variant)
- Fallback extraction of last path segment

### Authentication

Follows the Nebius Go SDK authentication hierarchy:
1. Service account key file (recommended for production)
2. IAM token (for testing/development)
3. Workload identity (automatic for MK8s pods)

## Testing

- [x] Unit tests pass (`go test ./pkg/csp/...`)
- [x] Integration tested on Nebius MK8s cluster with H200 GPUs
- [x] CSP client initializes successfully
- [x] Provider ID parsing validated against real cluster nodes

## Configuration Example

```yaml
janitor:
  csp:
    provider: "nebius"
  extraEnv:
    - name: NEBIUS_IAM_TOKEN
      valueFrom:
        secretKeyRef:
          name: nebius-janitor-token
          key: token
```

## Dependencies

- `github.com/nebius/gosdk` - Official Nebius Go SDK


<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

* **New Features**
  * Added Nebius (MK8s) as a supported cloud provider with configurable credentials, instance lifecycle actions (reboot/start/terminate), and readiness polling for node management.
* **Documentation**
  * Added Nebius-specific configuration guidance and environment-variable notes in deployment values.
* **Tests**
  * Extended tests to cover Nebius provider detection, environment parsing, and selection flows.
* **Chores**
  * Updated dependencies to support the Nebius integration.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

nv-oviya committed:
feat(janitor): add Nebius MK8s CSP support for node reboot operations using official Nebius Go SDK (github.com/nebius/gosdk) following same patterns as other CSP implementations
Signed-off-by: Oviya Seeniraj <oseeniraj@nvidia.com>

nv-oviya committed:
docs(janitor): add package-level godoc and improve docstring coverage for Nebius CSP

Signed-off-by: Oviya Seeniraj <oseeniraj@nvidia.com>

nv-oviya committed:
fix(janitor): address CodeRabbit review - remove false workload identity claim and use structured logging

Signed-off-by: Oviya Seeniraj <oseeniraj@nvidia.com>

lalitadithya commented on janitor/pkg/csp/nebius/nebius.go:
```diff
@@ -0,0 +1,279 @@
+	}
+
+	// Wait for the stop operation to complete
+	if _, err := op.Wait(ctx); err != nil {
```
Can we make this aync? I.e., can we store the op in `model.ResetSignalRequestRef` and then use `IsNodeReady` for polling? Please see `janitor/pkg/csp/gcp/gcp.go`, we have done something similar there

lalitadithya commented:
/ok to test 2ea70c8cea1c8493277b53ef872e2f206e692055

nv-oviya committed:
removed redundant logger.Error() calls, wrapped errands w/ context, made stop operation async like GCP pattern, and removed unused time import

Signed-off-by: Oviya Seeniraj <oseeniraj@nvidia.com>

nv-oviya commented:
/ok to test a650320

lalitadithya commented:
/ok to test a6503206dd8f801d46c9c8bbdfb3fceab9fea0b2

nv-oviya committed:
fix: resolve linting errors in nebius CSP client

- Fix import ordering (gofmt/goimports)
- Add exhaustive switch cases for InstanceStatus
- Add required whitespace before statements (wsl_v5)

Signed-off-by: Oviya Seeniraj <oseeniraj@nvidia.com>

nv-oviya committed:
fix: mark nebius/gosdk and grpc as direct dependencies

Signed-off-by: Oviya Seeniraj <oseeniraj@nvidia.com>

nv-oviya commented on janitor/go.mod:
```diff
@@ -99,6 +102,7 @@ require (
 	github.com/modern-go/concurrent v0.0.0-20180306012644-bacd9c7ef1dd // indirect
 	github.com/modern-go/reflect2 v1.0.3-0.20250322232337-35a7c28c31ee // indirect
 	github.com/munnerz/goautoneg v0.0.0-20191010083416-a7dc8b61c822 // indirect
+	github.com/nebius/gosdk v0.0.0-20251217093515-85d26a34ceb2 // indirect
```
Done, moved github.com/nebius/gosdk to the main require block as a direct dependency and ran go mod tidy -- the module is now properly listed as a direct dependency

nv-oviya committed:
fix: add terminal return path in IsNodeReady for unhandled states

Signed-off-by: Oviya Seeniraj <oseeniraj@nvidia.com>

nv-oviya committed:
fix: update dependencies to match main branch versions

Signed-off-by: Oviya Seeniraj <oseeniraj@nvidia.com>

nv-oviya committed:
fix: resolve merge conflicts and update protobuf to v1.36.11

Signed-off-by: Oviya Seeniraj <oseeniraj@nvidia.com>

nv-oviya committed:
Merge branch 'main' into oviya/add-nebius-janitor-csp

lalitadithya committed:
Merge branch 'main' into oviya/add-nebius-janitor-csp
